src/assets/TokenRatesController.ts

```diff
@@ -312,29 +380,90 @@ export class TokenRatesController extends BaseController<
     if (this.tokenList.length === 0 || this.disabled) {
       return;
     }
-    const { nativeCurrency } = this.config;
-
     const slug = await this.getChainSlug();
 
-    const newContractExchangeRates: ContractExchangeRates = {};
+    let newContractExchangeRates: ContractExchangeRates = {};
     if (!slug) {
       this.tokenList.forEach((token) => {
         const address = toChecksumHexAddress(token.address);
         newContractExchangeRates[address] = undefined;
       });
     } else {
-      const pairs = this.tokenList.map((token) => token.address).join(',');
-      const query = `contract_addresses=${pairs}&vs_currencies=${nativeCurrency.toLowerCase()}`;
-      const prices = await this.fetchExchangeRate(slug, query);
+      const { nativeCurrency } = this.config;
+      newContractExchangeRates = await this.fetchAndMapExchangeRates(
+        nativeCurrency,
+        slug,
+      );
+    }
+    this.update({ contractExchangeRates: newContractExchangeRates });
+  }
+
+  /**
+   * Checks if the active network's native currency is supported by the coingecko API
+   * If supported. fetches and maps contractExchange rates in format to be consumed by UI
+   * If not supported fetches contractExchange rates and maps from from token/fallback-currency to token/nativeCurrency
+   *
+   * @param nativeCurrency - the native currency of the currently active network
+   * @param slug - the unique slug used to id the chain by the coingecko api
+   * should be used to query token exchange rates
+   * @returns - Promise resolving to an object with conversion rates for each token
```

Comment:
Nit: the `@returns` directives should not include a dash. The dash is meant to separate the variable name from the description, but there is no variable name in the `@returns` directive.

```suggestion
   * @returns Promise resolving to an object with conversion rates for each token
```

Reply:
Also, the "Promise resolving to" part is implied by the type signature. This might be a better description:
```suggestion
   * @returns An object with conversion rates for each token
```